lib/aws_credentials.go

```diff
@@ -106,19 +113,27 @@ func (c *AWSCredentials) AssumeRole(arn string, duration time.Duration) (*AWSCre
 		return nil, err
 	}
 
-	return AWSCredentialsFromSTSCredentials(assumeRole.Credentials), nil
+	return AWSCredentialsFromSTSCredentials(assumeRole.Credentials, stsClient.Config.Region), nil
 }
 
 func (c *AWSCredentials) stsClient() (*sts.STS, error) {
-	// if c is nil, the default credential provider chain is used
+	// if c is nil, default configs are used
 	// (yes, I know this seems a little weird)
 	config := &aws.Config{}
-	if c != nil && c.ID != "" {
-		config.Credentials = credentials.NewStaticCredentials(
-			c.ID,
-			c.Secret,
-			c.Token,
-		)
+
+	if c != nil {
+		if c.Region != nil {
```

Comment:
You could push this nil check down a level if you made your custom resolver return the global endpoint when no region is set. I don't know if that gains us much but it's an option.

Reply:
Good call. This was originally written this way before I realized a custom resolver was necessary.